frame/collective/src/lib.rs

```diff
@@ -708,10 +730,128 @@ decl_module! {
 			let proposal_count = Self::do_disapprove_proposal(proposal_hash);
 			Ok(Some(T::WeightInfo::disapprove_proposal(proposal_count)).into())
 		}
+
+		/// Re-dispatch a collective origin call as a unique account that represents that origin.
+		///
+		/// Since the collective may want to be able to represent a different ratio than exists,
+		/// we allow the re-dispatch to specify the target ratio, as long as it is less restrictive
+		/// than the provided origin.
+		///
+		/// Can only be called by the collective origin.
+		#[weight = {
+			T::WeightInfo::dispatch_as_ratio_account(*length_bound)
+			.saturating_add(proposal.get_dispatch_info().weight)
+		}]
+		fn dispatch_as_ratio_account(origin,
+			proposal: Box<T::Proposal>,
+			#[compact] length_bound: u32,
+			#[compact] target_n: u32,
+			#[compact] target_d: u32,
+		) -> DispatchResult {
+			let maybe_raw_origin: Result<RawOrigin<T::AccountId, I>, _> = <T as Config<I>>::Origin::from(origin).into();
+
+			let raw_origin = maybe_raw_origin.map_err(|_| DispatchError::BadOrigin)?;
+
+			let proposal_len = proposal.using_encoded(|x| x.len());
+			ensure!(proposal_len <= length_bound as usize, Error::<T, I>::WrongProposalLength);
+
+			let account = Self::origin_to_ratio_account(raw_origin, target_n, target_d)?;
+			let proposal_hash = T::Hashing::hash_of(&proposal);
+			let signed_origin = frame_system::RawOrigin::Signed(account.clone());
+			let result = proposal.dispatch(signed_origin.into());
+			Self::deposit_event(
+				RawEvent::AccountExecuted(account, proposal_hash, result.map(|_| ()).map_err(|e| e.error))
+			);
+			Ok(())
+		}
+
+		/// Re-dispatch a collective origin call as a unique account that represents that origin.
+		///
+		/// Since the collective may want to be able to represent a different ratio than exists,
+		/// we allow the re-dispatch to specify the target ratio, as long as it is less restrictive
+		/// than the provided origin.
+		///
+		/// Can only be called by the collective origin.
+		#[weight = {
+			T::WeightInfo::dispatch_as_quantity_account(*length_bound)
+			.saturating_add(proposal.get_dispatch_info().weight)
+		}]
+		fn dispatch_as_quantity_account(origin,
+			proposal: Box<T::Proposal>,
+			#[compact] length_bound: u32,
+		) -> DispatchResult {
```

Comment:
Why does this one not have a `target_n` param?

Reply:
Should be controllable by the `propose` function which has a member threshold input?